src/ctapipe/reco/impact.py

```diff
@@ -85,51 +90,59 @@ class ImPACTReconstructor(Component):
 
     """
 
+    minimiser = traits.CaselessStrEnum(
+        ["minuit", "l-BFGS"],
+        default_value="minuit",
+        help="name minimiser to use in the fit",
+    ).tag(config=True)
+
+    use_time_gradient = traits.Bool(
+        default_value=False, help="Use time gradient in ImPACT reconstruction"
+    ).tag(config=True)
+
+    root_dir = traits.Unicode(
+        default_value=".", help="Directory containing ImPACT tables"
+    ).tag(config=True)
+
     # For likelihood calculation we need the with of the
     # pedestal distribution for each pixel
     # currently this is not available from the calibration,
     # so for now lets hard code it in a dict
     ped_table = {
-        "LSTCam": 2.8,
-        "NectarCam": 2.3,
-        "FlashCam": 2.3,
+        "LSTCam": 1.4,
+        "NectarCam": 1.3,
+        "FlashCam": 1.3,
+        "SST-Camera": 0.5,
         "CHEC": 0.5,
-        "DUMMY": 0,
+        "ASTRICam": 0.5,
+        "dummy": 0.01,
+        "UNKNOWN-960PX": 1.0,
     }
-    spe = 0.5  # Also hard code single p.e. distribution width
+    spe = 0.6  # Also hard code single p.e. distribution width
+
+    property = ReconstructionProperty.ENERGY | ReconstructionProperty.GEOMETRY
 
     def __init__(
-        self,
-        root_dir=".",
-        minimiser="minuit",
-        prior="",
-        template_scale=1.0,
-        xmax_offset=0,
-        use_time_gradient=False,
+        self, subarray, atmosphere_profile, dummy_reconstructor=False, **kwargs
     ):
         """
         Create a new instance of ImPACTReconstructor
         """
 
+        if atmosphere_profile is None:
+            raise TypeError(
+                "Argument 'atmosphere_profile' can not be 'None' for ImPACTReconstructor"
+            )
+
+        super().__init__(subarray, atmosphere_profile, **kwargs)
+
         # First we create a dictionary of image template interpolators
         # for each telescope type
-        self.root_dir = root_dir
-        self.priors = prior
-        self.minimiser_name = minimiser
-
-        self.file_names = {
-            "CHEC": ["GCT_05deg_ada.template.gz", "GCT_05deg_time.template.gz"],
-            "LSTCam": ["LST_05deg.template.gz", "LST_05deg_time.template.gz"],
-            "NectarCam": ["MST_05deg.template.gz", "MST_05deg_time.template.gz"],
-            "FlashCam": ["MST_xm_full.fits"],
-        }
-
-        # We also need a conversion function from height above ground to
-        # depth of maximum To do this we need the conversion table from CORSIKA
-        (
-            self.thickness_profile,
-            self.altitude_profile,
-        ) = get_atmosphere_profile_functions("paranal", with_units=False)
+        # self.priors = prior
+
+        # String templates for loading ImPACT templates
+        self.amplitude_template = Template("${base}/${camera}.template.gz")
```

Comment:
Is templating only by camera name what we want? Shouldn't we at least use the full `str(telescope_description)`? 

E.g. FlashCam on MST vs. FlashCam on HESS II

Reply:
Yes